### PR TITLE
PP-5199 Add provider states for transaction details tests

### PR DESCRIPTION
**Pull request by stephencdaly**
Add provider states so selfservice tests for the transaction details screen can be enabled.

### CODE REVIEW

File: src/test/java/uk/gov/pay/connector/pact/TransactionsApiContractTest.java [hunk 1]
```diff
@@ -268,10 +270,53 @@ public void createChargeWithDelayedCaptureTrueAndAwaitingCaptureRequestStatus(Ma
         setUpSingleCharge(gatewayAccountId, chargeId, chargeExternalId, ChargeStatus.AWAITING_CAPTURE_REQUEST, ZonedDateTime.now(), true);
     }
 
-    @State("a charge in state CREATED with accountId 42 and chargeExternalId 100 and metadata exists")
```
Comment (oswaldquek) [inline src/test/java/uk/gov/pay/connector/pact/TransactionsApiContractTest.java:308]: is this State deletable? is it not being defined at https://github.com/alphagov/pay-selfservice/blob/116bc92d11af8220781379cceac0261387a63649/test/unit/clients/connector_client/connector_get_transaction_details_test.js#L132? or is the intention to delete it now and is doing so forcing consumers to change appropriately in their builds?

File: src/test/java/uk/gov/pay/connector/pact/TransactionsApiContractTest.java [hunk 2]
```diff
@@ -268,10 +270,53 @@ public void createChargeWithDelayedCaptureTrueAndAwaitingCaptureRequestStatus(Ma
...
+                .withTransactionId("aGatewayTransactionId")
+                .withAmount(100)
+                .withStatus(ChargeStatus.CAPTURED)
+                .build();
+        dbHelper.addCharge(addChargeParams);
+        dbHelper.updateChargeCardDetails(addChargeParams.getChargeId(),
+                AuthCardDetailsFixture.anAuthCardDetails().build());
+    }
+    
+    @State("Gateway account 42 exists and has a charge with id abc123 and started and failed charge events")
```
Comment (oswaldquek) [inline src/test/java/uk/gov/pay/connector/pact/TransactionsApiContractTest.java]: A more valid English statement would be `Gateway account 42 exists and has a charge with id abc123 and has CREATED and AUTHORISATION_REJECTED charge events`

**Review by oswaldquek — CHANGES_REQUESTED** [review]

File: src/test/java/uk/gov/pay/connector/pact/TransactionsApiContractTest.java (hunk 2 above)
Comment (stephencdaly) [reply src/test/java/uk/gov/pay/connector/pact/TransactionsApiContractTest.java]: good idea, I'll change it

File: src/test/java/uk/gov/pay/connector/pact/TransactionsApiContractTest.java (hunk 1 above)
Comment (stephencdaly) [reply src/test/java/uk/gov/pay/connector/pact/TransactionsApiContractTest.java:308]: Ah I didn't remove it, I just changed the charge external ID from "100" to "abc123" just out of wanting to make it obvious that the ID is a string, although that's probably not a necessity.

Those tests in selfservice currently have the provider "selfservice-to-be" so aren't uploaded to the pact broker or run currently. I'll have a further pull request modifying the selfservice tests to be enabled and work.

**Review by oswaldquek — APPROVED** [review]

**Pull request merged**